cmd/kubelet/app/server.go

```diff
@@ -422,9 +422,16 @@ func run(s *options.KubeletServer, kubeDeps *kubelet.KubeletDeps) (err error) {
 
 		if s.CloudProvider != componentconfigv1alpha1.AutoDetectCloudProvider {
 			cloud, err = cloudprovider.InitCloudProvider(s.CloudProvider, s.CloudConfigFile)
-			if err != nil {
-				return err
+
+			switch err {
+			case cloudprovider.ErrNoConfig:
+				cloud = nil
+				glog.Warningf("Disabling cloud provider: %v", err)
```

Comment:
@derekwaynecarr this looks like a significant change for the kubelet.  You used to be guaranteed that you had a `cloud` if you continued.  Now you won't.  Looks risky.